src/platform/packages/private/kbn-grid-layout/grid/grid_panel/grid_panel.tsx

```diff
@@ -24,47 +24,14 @@ export interface GridPanelProps {
     panelId: string,
     setDragHandles?: (refs: Array<HTMLElement | null>) => void
   ) => React.ReactNode;
-  interactionStart: (type: PanelInteractionEvent['type'] | 'drop', e: UserInteractionEvent) => void;
   gridLayoutStateManager: GridLayoutStateManager;
 }
 
 export const GridPanel = forwardRef<HTMLDivElement, GridPanelProps>(
-  (
-    { panelId, rowIndex, renderPanelContents, interactionStart, gridLayoutStateManager },
-    panelRef
-  ) => {
+  ({ panelId, rowIndex, renderPanelContents, gridLayoutStateManager }, panelRef) => {
     const [dragHandleApi, setDragHandleApi] = useState<DragHandleApi | null>(null);
     const { euiTheme } = useEuiTheme();
 
-    useEffect(() => {
```

Comment:
With the current flow, this is not needed anymore. 

This is managed within the `useGridLayoutEvents` function, where the mouseup listener is attached to handle the event after mouse down. This way, the logic for interacting with the layout is streamlined and keeps all event management in one place.

Reply:
Is this removing that second render that caused the performance degradation?

Reply:
I made some investigation and it looks like we had 3 initial rerenders. This PR cuts one because we're removing `interactionStart` callback prop, but the reason for it to run in the first place is the cascade rerender after `renderPanelContents` is re-calculated because `dashboardContainer` dependency changed. Via using ref instead of the value of ref as a prop, I was able to get rid of 2 out of 3 re-renders (even without the changes from this PR). Check it out here: https://github.com/elastic/kibana/pull/208050